### PR TITLE
fix(filetype): check ft is emtpy

### LINKED ISSUE #20763: filetype: when open a new filetype file will get an error
### Neovim version (nvim -v)

NVIM v0.9.0-dev-143+ga288b4f21-dirty 

### Vim (not Nvim) behaves the same?

works fine

### Operating system/version

macos 12.6

### Terminal name/version

kitty

### $TERM environment variable

xterm-kitty

### Installation

brew

### How to reproduce the issue

- `nvim --clean`
- `:e test.db`

### Expected behavior

no error

### Actual behavior

error through

### PR DESCRIPTION
**Pull request by glepnir**
Fix #20763

### CODE REVIEW

**Comment by glepnir** [comment]
this check is wrong. should fix the `ft` type .

File: runtime/filetype.lua [hunk 1]
```diff
@@ -10,7 +10,15 @@ vim.api.nvim_create_autocmd({ 'BufRead', 'BufNewFile', 'StdinReadPost' }, {
...
-    if not ft then
+    local has_ft = false
+
+    if type(ft) == "nil" then
+      has_ft = not ft
+    elseif type(ft) == "string" then
+      has_ft = #ft == 0
+    end
+
+    if has_ft then
```
Comment (clason) [inline runtime/filetype.lua:21]:
```suggestion
   if not ft or ft == '' then
```

Comment (glepnir) [reply runtime/filetype.lua:21]: yep .can do this. but as i said the root of issue  we should explicitly give ft a type ? if you accept this check . i will update

**Comment by clason** [comment]
> but the root of the problem is that we should explicitly give ft a type.

I'd prefer not to touch the (relatively complex) detection logic at this point. My change fixes the issue without adding complexity.

**Comment by clason** [comment]
It'd be worth it finding out where `ft` is set to an empty string, though. That shouldn't happen, I think @smjonas ?

**Comment by clason** [comment]
I think a better fix would be to change this line
https://github.com/neovim/neovim/blob/24c9561a68aa9b9cf8d8a503c5e85f63d7ebb543/runtime/lua/vim/filetype.lua#L173
to just `bindzone(bufnr)`.

**Comment by smjonas** [comment]
> to just `bindzone(bufnr)`. Vim needs the empty string here, but in Lua we can pass `nil` just fine.

Yeah that seems like a good fix. Otherwise we can add a `if default ~= "" then` check just like in Vim

**Comment by clason** [comment]
>  Otherwise we can add a if default ~= "" then check just like in Vim

True. In this case, I'd prefer to diverge from Vim to be more idiomatic Lua as the change is very localized.

@glepnir Do you want to make that change to the `bindzone` call? Otherwise I can open a PR.

**Comment by glepnir** [comment]
I don't see the db extension :) yeah you can open a pr for that. my computer keeps restarting. I think what I'm going to do now is fix the computer

**Pull request closed without merging**